src/main/java/no/fint/betaling/service/FileService.java

```diff
@@ -0,0 +1,178 @@
+    }
+
+    public CustomerGroup getCustomers(String schoolId, ArrayList<String> customerNumbers) {
+        SkoleResource school = getSchool(schoolId);
+        CustomerGroup customerGroup = createCustomerGroup(school);
+        List<Customer> customerList = customerGroup.getCustomers().stream().filter(customer -> studentMatchStudentNumber(customer, customerNumbers)).collect(Collectors.toList());
+        customerGroup.setCustomers(customerList);
+        return customerGroup;
+    }
+
+    private boolean studentMatchStudentNumber(Customer customer, ArrayList<String> customerNumbers) {
+        return customerNumbers.contains(customer.getId());
+    }
+
+    private CustomerGroup createCustomerGroup(SkoleResource school) {
+        CustomerGroup customerGroup = new CustomerGroup();
+        customerGroup.setName(school.getNavn());
+        customerGroup.setDescription(null);
+        customerGroup.setCustomers(getCustomersForGroup(school.getElevforhold()));
+        return customerGroup;
+    }
+
+    private SkoleResource getSchool(String schoolId) {
+        return fileRepository.getSchools().values().stream()
+                .filter(hasOrganisationNumber(schoolId))
+                .findFirst()
+                .orElseThrow(() -> new SchoolNotFoundException(String.format("x-school-org-id: %s", schoolId)));
+    }
+
+    private Predicate<SkoleResource> hasOrganisationNumber(String schoolId) {
+        return s -> Optional.ofNullable(s.getOrganisasjonsnummer())
+                .map(Identifikator::getIdentifikatorverdi)
+                .map(schoolId::equals)
+                .orElse(false);
+    }
+
+    private List<Customer> getCustomersForGroup(List<Link> studentRelationLinks) {
+        if (studentRelationLinks == null) {
+            return Collections.emptyList();
+        }
+        Map<Link, ElevforholdResource> studentRelations = fileRepository.getStudentRelations();
+        Map<Link, PersonResource> students = fileRepository.getStudents();
+
+        return studentRelationLinks.stream()
+                .map(studentRelations::get)
+                .filter(Objects::nonNull)
+                .map(this::getStudentLink)
+                .filter(Objects::nonNull)
+                .map(students::get)
+                .filter(Objects::nonNull)
+                .map(CustomerFactory::toCustomer)
+                .distinct()
+                .collect(Collectors.toList());
+    }
+
+    private Link getStudentLink(ElevforholdResource resource) {
+        return resource.getElev().stream().findFirst().orElse(null);
+    }
+
+    public CustomerFileGroup getCustomersFromFile(String schoolId, byte[] file) throws UnableToReadFileException, NoVISIDColumnException {
+        InputStream targetStream = new ByteArrayInputStream(file);
+        Workbook wb;
+        try {
+            wb = WorkbookFactory.create(targetStream);
+        } catch (IOException ex) {
+            log.error(ex.getMessage(), ex);
```

Comment:
Avoid logging on `ERROR` level when the issue is handled and reported to the user anyway.

Reply:
avoided by changing to info